### PR TITLE
Built out redux framework with examples

### PR DESCRIPTION
**Pull request by kyang9021**
Most of the files in place can be deleted, they're currently there as an example for the basic framework for redux! 
Let me know if you have any questions on the code! 
@afong888 @chiakat @r-jeng Can one of you code review this when you have the time?

### CODE REVIEW

**Comment by kyang9021** [comment]
Oh btw, the last dispatch example fails the eslinter because I added a click handler to a static html element. This is just so you guys can see the state change in action when you click on the div!

File: client/src/components/ExampleProductList.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    isSuccess,
+    isError,
+    error,
+  } = useGetProductsQuery();
+
+  let content;
+
+  if (isLoading) {
+    content = (
+      <p>
```
Comment (chiakat) [inline client/src/components/ExampleProductList.jsx:17]: @kyang9021 I'm getting the warning` "Warning: Each child in a list should have a unique "key" prop."` Can you add `<p key={product.id}>` here?

Comment (kyang9021) [reply client/src/components/ExampleProductList.jsx:17]: Yup, Got it!

**Review by chiakat — APPROVED** [review]
@kyang9021 Looks good. Thanks for making all the changes and setting up the examples. I'll merge to main.

**Pull request merged**